### ISSUE TITLE
Finetuning pretrained models on a dataset other than Imagenet

### ISSUE THREAD

**Issue by KnockerPulsar** [facebookresearch/dino#116]
Hello. Firstly, thank you for the amazing work. 
Secondly, I'd like to use the already pretrained ViT models and finetune them on other (possibly unlabeled data). Specifically, I want to finetune the models to detect humans and segment them better. I know this probably has been asked before and I've read through issues similar to this one but I have a few questions:

1. I tried training the vit_small model on [this](https://www.kaggle.com/ifigotin/imagenetmini-1000) dataset. Since the models were pretrained on Imagenet-1k, they should do well on this dataset since it's a subset of that. However, the loss seems to be growing and not shrinking. I don't really have any idea why this happens.
2. How to train with unlabeled data? Do I just pass the model a folder with all the images inside or is there something else I need to do?
3. What about labeled data? What should be the structure of the training data? should it be `path/to/train/label/image.png`? or a different structure?

I'd appreciate if you guide me on any other things I'd need to finetune the pretrained models.

**Comment by woctezuma** [comment]
1. Not sure which loss you are talking about. This could be relevant: https://github.com/facebookresearch/dino/issues/61#issuecomment-904561965

**Comment by mathildecaron31** [comment]
Hi @KnockerPulsar 
Thanks for your kind words and your interest in this work. Sorry for the delay of my reply I've been taking some time off.

1. I don't think the fact that the loss is growing is necessarily a sign that the features are getting worse. If you look at our training logs provided in this repo you will see that it usually happens that the loss increases but the features actually improve.
2. 3. Yes you just need to adapt the dataset definition here https://github.com/facebookresearch/dino/blob/de9ee3df6cf39fac952ab558447af1fa1365362a/main_dino.py#L145 to your own usecase. For example you can take a look at https://pytorch.org/tutorials/beginner/data_loading_tutorial.html#dataset-class

You can take a look at for how to start fine-tuning from a pre-trained dino model: https://github.com/facebookresearch/dino/issues/80#issuecomment-880578855
